src/CacheManager.Core/Internal/BaseCache.cs

```diff
@@ -407,6 +410,55 @@ public virtual bool Remove(string key, string region)
         }
 
         /// <summary>
+        /// Gets all keys that match the given pattern.
+        /// </summary>
+        /// <param name="pattern">A glob to match against keys.</param>
+        /// <param name="region">The cache region.</param>
+        /// <returns>The matching unique keys in the cache</returns>
+        public virtual IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keys = AllKeys();
+            if (region != null)
+                keys = keys.Where(k => k.StartsWith(region + ":"));
+
+            if (pattern == "*")
+            {
+                return keys;
+            }
+            if (pattern.Contains("*") || pattern.Contains("?"))
+            {
+                var regexPattern = "^" + Regex.Escape(pattern).Replace(@"\*", ".*").Replace(@"\?", ".") + "$";
```

Comment:
Yup that's fine, lets have a starts/ends with implementation indicated by `*` at the beginning or end of the pattern? And if it gets more complex, fall back to a regex.